File: Find the Closest Element in BST.cpp
class Solution
{
    public:
    int minDiff(Node *root, int K)
    {
        if(root == NULL) return 1e9;
        if(root->data == K) return 0;
        int left = 1e9, right = 1e9;
        if(K < val) left = minDiff(root->left, K);
        else right = minDiff(root->right, K);
        return min(left, min(abs(K - val), right));
    }
};

File: June-2023/Find the Closest Element in BST.java
class Solution
{
    //Function to find the least absolute difference between any node
    //value of the BST and the given integer.
    static int minDiff(Node  root, int K) 
    { 
        // Write your code here
        int min = Integer.MAX_VALUE;
        Node curr = root;
        while(curr != null){
            int d = Math.abs(K-curr.data);
            if(d < min){
                min = d;
            }
            // else
            // break;
            
            if(K<curr.data)
            curr = curr.left;
            else
            curr = curr.right;
        }
        return min;
    } 
